tests/Integration/EnvironmentEncryptCommandTest.php

```diff
@@ -56,3 +56,59 @@
         ->expectsOutputToContain('Environment successfully encrypted.')
         ->assertExitCode(0);
 });
+
+
+it('prevents rotating unchanged already encrypted secrets', function () {
+    $plainContents = <<<'Text'
+        APP_KEY=1234
+        APP_NAME=Laravel
+        APP_ENV=local
+        APP_DEBUG=true
+        APP_URL=http://localhost
+        API_TOKEN=secret
+        NEW_SECRET=newvalue
+        CHANGED_SECRET=secret-2
+        Text;
+
+    $encrypter = new Encrypter('abcdefghijklmnopabcdefghijklmnop', 'AES-256-CBC');
+
+    $encryptedAppKey = $encrypter->encrypt('1234');
+    $encryptedApiToken = $encrypter->encrypt('secret');
+    $encryptedChangedSecret = $encrypter->encrypt('secret-1');
+
+    $existingEncryptedContents = implode(PHP_EOL, [
+        "APP_KEY={$encryptedAppKey}",
+        "API_TOKEN={$encryptedApiToken}",
+        "CHANGED_SECRET={$encryptedChangedSecret}",
+    ]);
+
+    $this->filesystem->shouldReceive('exists')
+        ->andReturn(true, true, true)
+        ->shouldReceive('get')
+        ->andReturn($plainContents, $existingEncryptedContents)
+        ->shouldReceive('put')
+        ->withArgs(function ($file, $contents) use ($encryptedAppKey, $encryptedApiToken) {
+            // The command should reuse existing encrypted values, not rotate them
+            $this->assertEquals($encryptedAppKey, Str::betweenFirst($contents, 'APP_KEY=', PHP_EOL));
+            $this->assertEquals($encryptedApiToken, Str::betweenFirst($contents, 'API_TOKEN=', PHP_EOL));
+
+            // The command should encrypt changed secrets
+            $this->assertNotEquals($encryptedAppKey, Str::betweenFirst($contents, 'CHANGED_SECRET=', PHP_EOL));
+
+            // The command should encrypt new secrets
+            $this->assertNotEquals($encryptedAppKey, Str::betweenFirst($contents, 'NEW_SECRET=', PHP_EOL));
```

Comment:
These two `assertNotEquals` validate the encrypted APP_KEY against CHANGED_SECRET AND NEW_SECRET. That doesn't make sense.

I think we could best validate that the `$content` has the correct value (by decrypting it in the test). That would prove that we didn't add a bug.
Something like this:
```
decryptedChangedSecret = $encrypter->decrypt($partOfContent);
$this->assertEquals($decryptedChangedSecret, 'secret-1');
```